pkg/scan/scanner.go

```diff
@@ -95,24 +95,15 @@ func (s *Scanner) Scan(ctx *tasks.ScanContext, stype ScanType, namespace, id str
 	results := make(chan []output.IOCFound)
 	defer close(results)
 
-	done := make(chan bool)
-	defer close(done)
-
 	m := [2][]output.IOCFound{}
 	i := 0
 
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		for {
-			select {
-			case malwares := <-results:
-				for _, malware := range malwares {
-					outputFn(malware, scanID)
-				}
-			case <-done:
-				logrus.Info("scan completed")
-				return
+		for malwares := range results {
```

Comment:
IMO, since results channel is not closed, this routine never finishes. so wg.Done() is not called.
wg.Wait() will keep waiting.

we need to add close(results) before wg.Wait()

Reply:
Indeed